src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/CpsDiagnosticItemSource.cs

```diff
@@ -8,115 +8,119 @@
 using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.Internal.VisualStudio.PlatformUI;
 using Microsoft.VisualStudio.Shell;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
+
+internal sealed partial class CpsDiagnosticItemSource : BaseDiagnosticAndGeneratorItemSource, INotifyPropertyChanged
 {
-    internal partial class CpsDiagnosticItemSource : BaseDiagnosticAndGeneratorItemSource, INotifyPropertyChanged
+    private readonly IVsHierarchyItem _item;
+    private readonly string _projectDirectoryPath;
+
+    public event PropertyChangedEventHandler? PropertyChanged;
+
+    public CpsDiagnosticItemSource(
+        IThreadingContext threadingContext,
+        Workspace workspace,
+        string projectPath,
+        ProjectId projectId,
+        IVsHierarchyItem item,
+        IAnalyzersCommandHandler commandHandler,
+        IDiagnosticAnalyzerService analyzerService,
+        IAsynchronousOperationListenerProvider listenerProvider)
+        : base(threadingContext, workspace, projectId, commandHandler, analyzerService, listenerProvider)
     {
-        private readonly IVsHierarchyItem _item;
-        private readonly string _projectDirectoryPath;
-
-        /// <summary>
-        /// The analyzer reference that has been found. Once it's been assigned a non-null value, it'll never be assigned null again.
-        /// </summary>
-        private AnalyzerReference? _analyzerReference;
```

Comment:
moved to base class.

Reply:
note that this comment is important.  it was an invariant of this subclass that it would only set this once.  We check for thsi invariant in the base class now.